### PR TITLE
Dev/delimiter s3sofs

### PR DESCRIPTION
**Pull request by jeremyds**

### CODE REVIEW

File: lib/algos/list/delimiter.js [hunk 1]
```diff
@@ -56,6 +62,8 @@ class Delimiter extends Extension {
         this.prefix = parameters.prefix;
         this.marker = parameters.marker;
         this.maxKeys = parameters.maxKeys || 1000;
+        this.alphabeticalOrder = parameters.alphabeticalOrder || true;
```
Comment (LaurenSpiegel) [inline lib/algos/list/delimiter.js]: won't this always be true since if parameters.alphaticalOrder is false or undefined this will return true?

Comment (LaurenSpiegel) [inline lib/algos/list/delimiter.js]: adding a unit test would be good

Comment (jeremyds) [inline lib/algos/list/delimiter.js]: test in progress to check the constructor and the filter method behavior.

Comment (jeremyds) [inline lib/algos/list/delimiter.js]: done

File: tests/unit/algos/list/delimiter.js [hunk 2]
```diff
@@ -274,6 +306,18 @@ describe('Delimiter listing algorithm', () => {
         done();
     });
 
+    it('Should set Delimiter alphabeticalOrder field to the good value',
```
Comment (LaurenSpiegel) [inline tests/unit/algos/list/delimiter.js]: 'Should set Delimiter alphabeticalOrder field to the expected value'

File: tests/unit/algos/list/delimiter.js [hunk 3]
```diff
@@ -274,6 +306,18 @@ describe('Delimiter listing algorithm', () => {
...
+       done =>
```
Comment (LaurenSpiegel) [inline tests/unit/algos/list/delimiter.js]: this is not async so don't need done

File: tests/unit/algos/list/delimiter.js (hunk 2 above)
Comment (jeremyds) [inline tests/unit/algos/list/delimiter.js]: done

File: tests/unit/algos/list/delimiter.js (hunk 3 above)
Comment (jeremyds) [inline tests/unit/algos/list/delimiter.js]: done

Comment (LaurenSpiegel) [inline tests/unit/algos/list/delimiter.js]: ironic 😄

**Comment by jeremyds** [comment]
@ironman-machine go

**Pull request merged**